back/src/Application/Buying/Handler/CreateBuyingCommandHandler.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Application\Buying\Handler;
+
+use App\Application\Buying\Command\CreateBuyingCommand;
+use App\Domain\Apiary\Repository\ApiaryRepositoryInterface;
+use App\Domain\Buying\Buying;
+use App\Domain\Buying\Repository\BuyingRepositoryInterface;
+
+class CreateBuyingCommandHandler
+{
+    public function __construct(
+        private readonly BuyingRepositoryInterface $buyingRepository,
+        private readonly ApiaryRepositoryInterface $apiaryRepository,
+    ) {
+    }
+
+    public function __invoke(CreateBuyingCommand $command): Buying
+    {
+        $payload = $command->payload;
+        $user = $command->user;
+        // $apiary = $command->apiary;
+
+        $apiary = $this->apiaryRepository->getOneByUid($payload->apiary);
+        if ($payload->apiary !== null) {
+            $apiary = $this->apiaryRepository->getOneByUid($payload->apiary);
+        }
+
+        $buying = new Buying(
+            $payload->label,
+            $payload->price,
+            $payload->date,
+            $apiary, //  null si non applicable
```

Comment:
```suggestion
            $apiary, 
```
Après c'est à ta convenance, mais je suis un peu partisane de commenter quand c'est strictement nécessaire, là on décrit déjà plus haut que apiary peut-être null vu que l'on l'initie avec cette valeur (cf. review)

Reply:
Toujours UP